lib/Client.js

```diff
@@ -24,28 +23,50 @@ function Client(config) {
   // events later on.
   events.EventEmitter.call(self);
 
-  // Bind the user-supplied config property to our class so it can be accessed
-  // like an object.
-  self.config = new Config(config);
-
-  // For backwards compatibility reasons, if no API key is specified we'll try
-  // to grab the API credentials out of our new format and shove it into the old
-  // format.  This can go away once we cut a release and decide to no longer
-  // support the old configuration formatting.
-  if (!self.config.apiKey) {
-    self.config.apiKey = self.config.client.apiKey;
-  }
-
-  self._dataStore = new DataStore(self.config);
-
   // We'll maintain this class variable as an in-memory singleton for caching
   // purposes.  We do this because Tenants never ever change once a Client has
   // been initialized, so it makes sense to cache the Tenant object so we don't
   // make unnecessary API requests if this object is looked up more than once.
   self._currentTenant = null;
 
-  self.mergeRemoteConfig(function(err) {
-    return err ? self.emit('error', err): self.emit('ready', self);
+  // Indicates whether or not this client is ready yet.
+  self._isReady = false;
+
+  // Setup how we load our configuration.
+  var configLoader = null;
```

Comment:
Why are you setting this to `null`? Variables will be `undefined` if not assigned. Besides, you're still assigning it in the block below.